unit_12/main.js

```diff
@@ -0,0 +1,64 @@
+// Exercise 1
+const compareArray = (array1, array2) => {
+  if (JSON.stringify(array1.sort()) === JSON.stringify(array2.sort())) {
+    return true
+  } else  {
+    return false
+  } 
+}
+// Exercise 2:
+const newArray = []
+const flattenArray = array => {
+  for (let i = 0; i < array.length; i++) {
+    if (Array.isArray(array[i])) {
+      flattenArray(array[i])
+    } else {
+      newArray.push(array[i])
+    }
+  }
+  return newArray
+}
+
+// Exercise 3:
+const chunksArray = (array, size) => {
+  const newArray = []
+  while (array.length) {
+    newArray.push(array.splice(0, size))
+  }
+  return newArray
+}
+
+// Exercise -4
+function ResultArray(array1, array2) {
+  const result = []
+  for (let i = 0; i < array1.length; i++) {
+    for (let j = 0; j < array2.length; j++) {
+      if (array1[i] === array2[j]) {
+        result.push(array1[i])
+      }
+    }
+  }
+  return result
+}
+
+// Exercise 5:
+const checkPriceDownCode = date => {
+  const discountTime = new Date('Oct 30, 2020 12:30:32')
+  if (Date.parse(date) <= Date.parse(discountTime)) {
+    return true
+  } else {
+    return false
+  }
```

Comment:
```suggestion
  return Date.parse(date) <= Date.parse(discountTime)
```
Không viết các lệnh dạng
```js
if (true) {
  return true
} else {
  return false
}
```